src/expr/src/expr/expr_udf.rs

```diff
@@ -43,11 +47,27 @@ impl Expression for UdfExpression {
     }
 
     fn eval(&self, input: &DataChunk) -> Result<ArrayRef> {
-        todo!("evaluate UDF")
+        let columns: Vec<_> = self
+            .children
+            .iter()
+            .map(|c| c.eval_checked(input).map(|a| ("", a.as_ref().into())))
+            .try_collect()?;
+        let input =
+            arrow_array::RecordBatch::try_from_iter(columns).expect("failed to build record batch");
+        let output = tokio::task::block_in_place(|| {
+            tokio::runtime::Handle::current().block_on(self.client.call(&self.function_id, input))
+        })?;
```

Comment:
I'm curious about the future plan for making this non-blocking. 🥵 Will we make the expression framework asynchronous?

Reply:
Good question. 🥵 I found it's a big work to refactor all build and evaluation functions into async. So I decided to leave it blocked here for now. I'm not sure whether this is acceptable. Because according to the tokio document, `block_in_place` will not cause starvation of other tasks in a multithreaded executor. But I believe sooner or later we will make them async.